CMakeLists.txt

```diff
@@ -105,13 +98,30 @@ add_feature_info(WITH_SANITIZERS WITH_SANITIZERS "Build with address sanitizer a
 add_feature_info(WITH_MSAN WITH_MSAN "Build with memory sanitizer")
 add_feature_info(WITH_FUZZERS WITH_FUZZERS "Build test/fuzz")
 add_feature_info(WITH_NEW_STRATEGIES WITH_NEW_STRATEGIES "Use new strategies")
-if(BASEARCH_ARM_FOUND)
-    add_feature_info(WITH_ACLE WITH_ACLE "Build with ACLE CRC")
-    add_feature_info(WITH_NEON WITH_NEON "Build with NEON intrinsics")
-endif()
 add_feature_info(WITH_MAINTAINER_WARNINGS WITH_MAINTAINER_WARNINGS "Build with project maintainer warnings")
 add_feature_info(WITH_CODE_COVERAGE WITH_CODE_COVERAGE "Enable code coverage reporting")
 
+if(BASEARCH_ARM_FOUND)
+    set(WITH_ACLE "Build with ACLE" ON)
+    set(WITH_NEON "Build with NEON intrinsics" ON)
+    mark_as_advanced(WITH_ACLE)
+    mark_as_advanced(WITH_NEON)
+elseif(BASEARCH_S360_FOUND AND "${ARCH}" MATCHES "s390x")
+    set(WITH_DFLTCC_DEFLATE "Use DEFLATE CONVERSION CALL instruction for compression on IBM Z" OFF)
+    set(WITH_DFLTCC_INFLATE "Use DEFLATE CONVERSION CALL instruction for decompression on IBM Z" OFF)
+    mark_as_advanced(WITH_DFLTCC_DEFLATE)
+    mark_as_advanced(WITH_DFLTCC_INFLATE)
+elseif(BASEARCH_X86_FOUND)
+    set(WITH_AVX2 "Build with AVX2" ON)
+    set(WITH_SSE2 "Build with SSE2" ON)
+    set(WITH_SSE4 "Build with SSE4" ON)
+    set(WITH_PCLMULQDQ "Build with PCLMULQDQ" ON)
+    mark_as_advanced(WITH_AVX2)
+    mark_as_advanced(WITH_SSE2)
+    mark_as_advanced(WITH_SSE4)
+    mark_as_advanced(WITH_PCLMULQDQ)
+endif()
```

Comment:
All the above need to be option() not set(), otherwise they never show up in ccmake, even in advanced mode.

mark_as_advanced can mark multiple options as advanced.
According to the documentation of mark_as_advanced, it should either be used before making the option, or with the FORCE parameter to change it after the fact. I have not been able to reproduce any ill effects AFAIK, but I know there are potential problems with cache there.

I suggest adding this after the group of options.
```
    option(..)
    mark_as_advanced(FORCE WITH_AVX2 WITH_SSE2 WITH_SSE4 WITH_PCLMULQDQ)
```

Reply:
Ok let me know what you think now.

Reply:
Unless you know something I don't, you are missing the FORCE parameter. According to how I understood the docs, it is required when setting it after creating the options to avoid problems with it getting set as a new empty variable due to weirdness in the caching.
I think they might have fixed it in some of the most recent versions (saw a ticket about it from last year I think it was).

Reply:
Sorry I must have missed it. I’ve never used the mark_as_advanced macro before.

Reply:
Fixed.